lib/connectors/memory.js

```diff
@@ -251,6 +251,37 @@ Memory.prototype.updateOrCreate = function(model, data, options, callback) {
   });
 };
 
+Memory.prototype.patchOrCreateWithWhere =
+  Memory.prototype.upsertWithWhere = function(model, where, data, options, callback) {
+    var self = this;
+    var filter = where;
+    var primaryKey = this.idName(model);
+    var nodes = self._findAllSkippingIncludes(model, filter);
+
+    if (nodes.length === 0) {
+      return self._createSync(model, data, function(err, id) {
+        if (err) return process.nextTick(function() { cb(err); });
+        self.saveToFile(id, function(err, id) {
+          self.setIdValue(model, data, id);
+          callback(err, self.fromDb(model, data), { isNewInstance: true });
+        });
+      });
+    } else if (nodes.length === 1) {
+      var primaryKeyValue = nodes[0][primaryKey];
+      self.updateAttributes(model, primaryKeyValue, data, options, function(err, data) {
+        self.save(model, data, options, function(err, data) {
```

Comment:
I think `self.save` is not needed here - it's already called by `updateAttributes` under the hood.